First test PR to try out CI

sweverett opened the pull request:
This PR adds the basic models for intensity & velocity, as well as lots of utility methods, likelihood tests, first unit tests, environment updates, etc.

Sorry for whoever has to review this one!

pranjalrs commented on kl_pipe/intensity.py:
```diff
@@ -0,0 +1,105 @@
+from kl_pipe.model import IntensityModel
+
+
+class InclinedExponentialModel(IntensityModel):
```
This is simply an exponential right? I think an inclined exponential should include a vertical intensity/density distribution. E.g. galsim uses sech^2(z/z0)

sweverett commented on kl_pipe/velocity.py:
```diff
@@ -0,0 +1,176 @@
+        Circular velocity.
+    rscale : float
+        Scale radius.
+    sini : float
```
Sure, will make that change

sweverett reviewed (commented):
Did a sanity checking review and didn't see anything deeply offensive or obviously wrong. I recommend we merge this in ASAP so we can move on to the new TNG datavectors PR I'm about to make, and then a new branch to add more realistic data features, MCMC sampler integration, etc.

JiachuanXu commented on kl_pipe/synthetic.py:
```diff
@@ -0,0 +1,855 @@
+    profile = profile.rotate(theta_int * gs.radians)
+
+    # Apply shear and centroid offset
+    profile = profile.shear(g1=g1, g2=g2)
```
Inconsistent definition of shear between `GalSim` backend (this line) and `scipy` backend methods (L157 and L289). [The `shear` method](https://galsim-developers.github.io/GalSim/_build/html/shear.html#galsim.Shear.getMatrix) in `GalSim` conserves the area, while the `scipy` methods includes a magnification. The impact is small, but I think it would be good to make them consistent and mention the difference explicitly in code comments. I suggest adapting the `GalSim` backend method to match the `scipy` backend methods, i.e. use [the `lens` method ](https://github.com/GalSim-developers/GalSim/blob/releases/2.7/galsim/gsobject.py#L977)`profile.lens(g1=g1, g2=g2, mu=1/(1-g1*g1-g2*g2))`.

JiachuanXu commented on kl_pipe/transformation.py:
```diff
@@ -0,0 +1,220 @@
+    Parameters
+    ----------
+    theta_int : float
+        Intrinsic position angle.
```
I suggest mentioning the range of theta_int (and also cosi later) in the docstring. If I understand correctly, the definition of inclination angle is within [0, pi/2) and theta_int is within [0, 2*pi), since cos(inc) is equivalent to cos(-inc) with theta_int changed by pi.

JiachuanXu reviewed (approved):
I went through the PR, generally the code structure looks good to me. I don't have major feedbacks, except some minor ones (e.g. definition of shear transformation, pipeline homepage, user customized `base` env, etc.).

sweverett commented on README.md:
````diff
@@ -1,11 +1,147 @@
+
+```bash
+# Prerequisites (one-time setup)
+conda install -n base conda-lock  # If not already installed
````
Ok let's iterate on the solution on that issue page

sweverett commented on pyproject.toml:
oops! thanks

sweverett commented:
@JiachuanXu if you go to the files changed tab you can look at the commits since your last review if you want to check my code that addresses your comments

sweverett commented:
Comments addressed so I'll go ahead and merge now. Congrats all!
